e2e2/test/cases/netpol/main_test.go

```diff
@@ -0,0 +1,425 @@
+package netpol
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	networking "k8s.io/api/networking/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/third_party/helm"
+)
+
+var testenv env.Environment
+
+func TestMain(m *testing.M) {
+	cfg, err := envconf.NewFromFlags()
+	if err != nil {
+		log.Fatalf("failed to initialize test environment: %v", err)
+	}
+	testenv = env.NewWithConfig(cfg)
+	namespaces := []string{"a", "b", "c"}
+
+	testenv.Setup(
+
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+			if err != nil {
+				return ctx, err
+			}
+
+			servers := map[string]string{
+				"a": "a-server",
+				"b": "b-server",
+				"c": "c-server",
+			}
+
+			// 1. Install Latest CNI version
+			log.Print("Install the latest VPC-CNI on the cluster")
+			err = installLatestCNI(config, ctx)
+			if err != nil {
+				log.Print("Failed to install latest aws-vpc-cni on cluster")
+				return ctx, err
+			}
+
+			// 2. Create three namespaces
+			log.Print("Creating the test namespaces")
+			for _, ns := range namespaces {
+				err = createNamespace(ns, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create namespace %s", ns)
+					return ctx, err
+				}
+			}
+
+			// 3. Create deployment and service
+			log.Print("Creating the test deployment and service")
+			for ns, server := range servers {
+				err = createServerAndService(ns, server, 1, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create deployment and service for %s", server)
+					return ctx, err
+				}
+			}
+
+			return ctx, nil
+		},
+	)
+
+	testenv.Finish(
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+
+			if err != nil {
+				return ctx, err
+			}
+			log.Print("Deleting the test namespaces")
+			for _, ns := range namespaces {
+				client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns, Namespace: ns}})
+			}
+
+			return ctx, nil
+		},
+	)
+
+	os.Exit(testenv.Run(m))
+}
+
+func installLatestCNI(config *envconf.Config, ctx context.Context) error {
+
+	repoName := "eks"
+	repoUrl := "https://aws.github.io/eks-charts"
+	chartName := "eks/aws-vpc-cni"
+	releaseNamespace := "kube-system"
+
+	client, err := config.NewClient()
+
+	if err != nil {
+		log.Print("Failed to get client")
```

Comment:
Instead of logging before returning errors throughout this function, you could use `errors.Wrap` if the added context is really valuable